### PR TITLE
Added Let's Encrypt clarification 

### PR DESCRIPTION
**Pull request by icelander**
Added clarification that Let's Encrypt needs to be able to connect to your server to verify the domain

Signed-off-by: Paul Rothrock <paul@movetoiceland.com>

### CODE REVIEW

File: source/install/config-tls-mattermost.rst [hunk 1]
```diff
@@ -8,6 +8,9 @@ You have two options if you want users to connect with HTTPS:
 
 The easiest option is to set up TLS on the Mattermost Server, but if you expect to have more than 200 users, use a proxy for better performance. A proxy server also provides standard HTTP request logs.
 
+.. note::
+   Your Mattermost server must be accessible from the Let's Encrypt in order to verify your domain name and issue the certificate. Be sure to open your firewall and configure any reverse proxies to forward traffic to port 80 and 443. More information can be found `at Let's Encrypt <https://letsencrypt.org/how-it-works/>`__
```
Comment (amyblais) [inline source/install/config-tls-mattermost.rst]:
```suggestion
   Your Mattermost server must be accessible from the Let's Encrypt in order to verify your domain name and issue the certificate. Be sure to open your firewall and configure any reverse proxies to forward traffic to ports 80 and 443. More information can be found _`at Let's Encrypt <https://letsencrypt.org/how-it-works/>`__.
```

**Comment by jasonblais** [comment]
@crspeller can you help with a dev review, and then we can get this merged (after the one typo fix noted by Amy above)

File: source/install/config-tls-mattermost.rst (hunk 1 above)
Comment (crspeller) [inline source/install/config-tls-mattermost.rst]:
```suggestion
   Your Mattermost server must be accessible from the Let's Encrypt CA in order to verify your domain name and issue the certificate. Be sure to open your firewall and configure any reverse proxies to forward traffic to port 80 and 443. More information can be found `at Let's Encrypt <https://letsencrypt.org/how-it-works/>`__
```

**Comment by jasonblais** [comment]
Thanks! @icelander FYI on two small suggestions above, looks like I don't have push access to your branch to make them myself,

**Comment by jasonblais** [comment]
Friendly reminder to help with the above two suggestions @icelander when you have a moment :)

Otherwise looks like this is ready to merge

**Comment by justinegeffen** [comment]
Following up on this - @icelander, if you have a moment to address the two changes then it looks like we should be good to go.

File: source/install/config-tls-mattermost.rst [hunk 2]
```diff
@@ -8,6 +8,9 @@ You have two options if you want users to connect with HTTPS:
 
 The easiest option is to set up TLS on the Mattermost Server, but if you expect to have more than 200 users, use a proxy for better performance. A proxy server also provides standard HTTP request logs.
 
+.. note::
+   Your Mattermost server must be accessible from the Let's Encrypt in order to verify your domain name and issue the certificate. Be sure to open your firewall and configure any reverse proxies to forward traffic to ports 80 and 443. More information can be found _`at Let's Encrypt <https://letsencrypt.org/how-it-works/>`__.
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst]:
```suggestion
Your Mattermost server must be accessible from Let's Encrypt in order to verify your domain name and issue the certificate. Be sure to open your firewall and configure any reverse proxies to forward traffic to ports 80 and 443. More information can be found _`at Let's Encrypt <https://letsencrypt.org/how-it-works/>`__.
```

File: source/install/config-tls-mattermost.rst [hunk 3]
```diff
@@ -8,6 +8,9 @@ You have two options if you want users to connect with HTTPS:
...
+
 **Configure TLS on the Mattermost Server**:
 
 1. In the **System Console** > **General** > **Configuration** in prior versions or **System Console** > **Environment** > **Web Server** in versions after 5.12.
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst]:
```suggestion
1. In **System Console > Environment > Web Server** (or **System Console > General > Configuration** in versions prior to 5.12).
```

**Comment by justinegeffen** [comment]
Hi @icelander. I've committed the suggested changes as they were related to formatting. Re-requesting reviews so we can move the PR forward.

File: source/install/config-tls-mattermost.rst [hunk 4]
```diff
@@ -8,9 +8,12 @@ You have two options if you want users to connect with HTTPS:
 
 The easiest option is to set up TLS on the Mattermost Server, but if you expect to have more than 200 users, use a proxy for better performance. A proxy server also provides standard HTTP request logs.
 
+.. note::
+Your Mattermost server must be accessible from the Let's Encrypt CA in order to verify your domain name and issue the certificate. Be sure to open your firewall and configure any reverse proxies to forward traffic to ports 80 and 443. More information can be found _`at Let's Encrypt <https://letsencrypt.org/how-it-works/>`__.
```
Comment (amyblais) [inline source/install/config-tls-mattermost.rst:12]:
```suggestion
   Your Mattermost server must be accessible from the Let's Encrypt CA in order to verify your domain name and issue the certificate. Be sure to open your firewall and configure any reverse proxies to forward traffic to ports 80 and 443. More information can be found _`at Let's Encrypt <https://letsencrypt.org/how-it-works/>`_.
```

**Pull request merged**